docs/proposals/third-party-api-servers.md

```diff
@@ -0,0 +1,175 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/proposals/third-party-api-servers.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Third Party API Servers
+
+## Goal
+
+With third party API servers, we want to enable cluster admins or users to
```

Comment:
I do not believe we *must* have a reverse proxy for either.  If you use the
platform for hosting these components, you have the necessary tools to make
them HA (make the address registered in this API HA)

On Tue, Nov 17, 2015 at 5:43 PM, Brian Grant <notifications@github.com>
wrote:

> In docs/proposals/third-party-api-servers.md
> <https://github.com/kubernetes/kubernetes/pull/17191#discussion_r45133939>
> :
>
> > +[here](http://releases.k8s.io/release-1.0/docs/proposals/third-party-api-servers.md).
> > +
> > +Documentation for other releases can be found at
> > +[releases.k8s.io](http://releases.k8s.io).
> > +</strong>
> > +--
> > +
> > +<!-- END STRIP_FOR_RELEASE -->
> > +
> > +<!-- END MUNGE: UNVERSIONED_WARNING -->
> > +
> > +# Third Party API Servers
> > +
> > +## Goal
> > +
> > +With third party API servers, we want to enable cluster admins or users to
>
> And do we need a reverse proxy, anyway, for HA and scalability?
>
> And how does this relate to Ubernetes-style federation?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/17191/files#r45133939>.
>